internal/backup/volumesnapshot_action.go

```diff
@@ -60,8 +60,18 @@ func (p *VolumeSnapshotBackupItemAction) Execute(item runtime.Unstructured, back
 		return nil, nil, errors.WithStack(err)
 	}
 
+	// determine if we are backing up a volumesnapshot that was created by velero while performing backup of a
+	// CSI backed PVC.
+	// For volumesnapshots that were created during the backup of a CSI backed PVC, we will wait for the volumecontents to
+	// be available.
+	// For volumesnapshots created outside of velero, we expect the volumesnapshotcontent to be available prior to backing up
+	// the volumesnapshot. In case of a failure, backup should be re-attempted after the CSI driver has reconciled the volumesnapshot.
+	// existence of the velerov1api.BackupNameLabel indicates that the volumesnapshot was created while backing up a
+	// CSI backed PVC.
+	_, toWait := vs.Labels[velerov1api.BackupNameLabel]
```

Comment:
I think we also want to check if the snapshot was created for **this** Velero backup, not just any backup. So need to compare the label value to `backup.Name`. 

Unfortunately, backup name isn't foolproof here, since it's not unique over time (backup `foo` could be created, deleted, and recreated). We may need to use the backup UID, which will be unique over time. Need to think about this a little more, since using UID comes with its own set of gotchas.

For now, let's at least check the name and add a `// TODO `.

Reply:
done